frontend/public/components/project-overview.jsx

```diff
@@ -0,0 +1,341 @@
+      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
+      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
+      // Fall back to sorting by name if no deployment versions.
+      let leftName, rightName;
+      if (!_.isFinite(leftVersion) && !_.isFinite(rightVersion)) {
+        leftName = _.get(left, 'metadata.name', '');
+        rightName = _.get(right, 'metadata.name', '');
+        if (descending) {
+          return rightName.localeCompare(leftName);
+        }
+        return leftName.localeCompare(rightName);
+      }
+
+      if (!leftVersion) {
+        return descending ? 1 : -1;
+      }
+
+      if (!rightVersion) {
+        return descending ? -1 : 1;
+      }
+
+      if (descending) {
+        return rightVersion - leftVersion;
+      }
+
+      return leftVersion - rightVersion;
+    };
+
+    return _.toArray(replicationControllers).sort(compareDeployments);
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+    if (!filterValue) {
+      return items;
+    }
+    const filterString = filterValue.toLowerCase();
+    let filteredItems = [];
+    _.forEach(items, item => {
+      const name = _.get(item.deploymentConfig, 'metadata.name', '');
+      if (name.toLowerCase().includes(filterString)) {
+        filteredItems.push(item);
+      }
+    });
+
+    return filteredItems;
+  }
+
+  createTopologyData() {
+    const {loaded, replicationControllers, pods, deploymentConfigs, routes, services} = this.props;
+    let items = [];
+    if (!loaded) {
+      return;
+    }
+    _.forEach(replicationControllers.data, function (replicationController) {
+      const controllerUID = _.get(replicationController, 'metadata.uid');
+      replicationController.pods = _.filter(pods.data, function (pod) {
+        return _.some(_.get(pod, 'metadata.ownerReferences'), {
+          uid: controllerUID,
+          controller: true
+        });
+      });
+    });
+
+    let allServices = _.keyBy(services.data, 'metadata.name');
+
+    _.forEach(deploymentConfigs.data, (deploymentConfig) => {
+      deploymentConfig.kind = 'DeploymentConfig';
+      // Determine the replication controllers associated with this DC
+      const dcUID = _.get(deploymentConfig, 'metadata.uid');
+      const dcControllers = _.filter(replicationControllers.data, function (replicationController) {
+        const {ownerReferences} = replicationController.metadata;
+        return _.some(ownerReferences, {
+          uid: dcUID,
+          controller: true
+        });
+      });
+      const ordered = this.sortByDeploymentVersion(dcControllers, true);
+      deploymentConfig.replicationControllers = dcControllers;
+      deploymentConfig.currentController = _.head(ordered);
+      deploymentConfig.prevController = _.size(ordered) < 2 ? null : ordered[1];
+
+      // Get the services for this deployment config
+      let configServices = [];
```

Comment:
Nit, I'd suggest a `const configServices = _.reduce(allServices, () => {}, [])` to make these lists.